libs/model/src/policies/utils/community-indexer-utils.ts

```diff
@@ -0,0 +1,268 @@
+
+/**
+ * Asynchronously paginates through Clanker tokens.
+ *
+ * This function fetches tokens from the Clanker API in pages and yields batches of tokens that were
+ * created after the specified cutoff date. It continues fetching pages until one of the following conditions is met:
+ *
+ * - The API indicates there are no more pages available.
+ * - A fetched page returns no tokens.
+ * - The oldest token in the fetched page is older than the cutoff date.
+ *
+ * An axios instance with retry logic is used to handle rate limits (HTTP 429) and server errors (HTTP status >= 500).
+ *
+ * @param {Object} params - The function parameters.
+ * @param {Date} params.cutoffDate - The cutoff date; only tokens created after this date are yielded.
+ * @param {boolean} params.desc - If true, tokens are fetched in descending order; if false, in ascending order.
+ * @returns {AsyncGenerator<Array<z.infer<typeof ClankerToken>>, void, unknown>} An async generator that yields arrays of tokens meeting the cutoff criteria.
+ *
+ * @throws Will throw an error if the token fetching fails after retrying.
+ */
+export async function* paginateClankerTokens({
+  cutoffDate,
+  desc,
+}: {
+  cutoffDate: Date;
+  desc: boolean;
+}): AsyncGenerator<Array<z.infer<typeof ClankerToken>>, void, unknown> {
+  const axiosInstance = axios.create();
+  axiosRetry(axiosInstance, {
+    retries: 3,
+    retryDelay: (retryCount: number) => 5000 * Math.pow(2, retryCount - 1),
+    retryCondition: (error: AxiosError) =>
+      error.response?.status === 429 || (error.response?.status ?? 0) >= 500,
+  });
+
+  let pageNum = 1;
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const url = `https://www.clanker.world/api/tokens?sort=${desc ? 'desc' : 'asc'}&page=${pageNum}&pair=all&partner=all&presale=all`;
+    log.debug(`fetching: ${url}`);
+
+    try {
+      // Use the axios instance with retry logic
+      const res = await axiosInstance.get<{
+        data: Array<z.infer<typeof ClankerToken>>;
+        hasMore: boolean;
+      }>(url);
+
+      // If no tokens were returned, we're done
+      if (!res.data.hasMore || res.data.data.length === 0) {
+        log.debug('No more tokens found');
+        break;
+      }
+
+      // Filter tokens by cutoffDate
+      const validTokens = res.data.data.filter((t) =>
+        moment(t.created_at).isAfter(cutoffDate),
+      );
+
+      // Yield valid tokens
+      if (validTokens.length > 0) {
+        yield validTokens;
+      }
+
+      // Check if the oldest token is older than our cutoff date
+      const oldestToken = res.data.data[res.data.data.length - 1];
+      if (moment(oldestToken.created_at).isBefore(cutoffDate)) {
+        break;
+      }
+
+      // Move to the next page
+      pageNum++;
+    } catch (err: unknown) {
+      log.error(`Error fetching clanker tokens: ${(err as Error).message}`);
+      throw err;
+    }
+  }
+}
+
+function formatCommunityName(input: string) {
```

Comment:
We have `COMMUNITY_NAME_REGEX`. Are these definitions compatible? Should we add a check with that definition here?